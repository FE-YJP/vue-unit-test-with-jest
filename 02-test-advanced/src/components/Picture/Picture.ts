import Vue from 'vue'

export default Vue.extend({
  data: () => ({
    isShow: false
  }),

  methods: {
    changeShowLazy () {
      setTimeout(() => {
        this.isShow = !this.isShow
      }, 1000)
    }
  }
})
