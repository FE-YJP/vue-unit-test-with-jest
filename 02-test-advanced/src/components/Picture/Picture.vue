<template>
  <div>
    <v-btn
    @click="changeShowLazy"
    class="mt-4"
    large
    depressed
    color="warning"
   >
   点击一秒后显示/隐藏图片（定时器测试用例）
   </v-btn>
    <v-img v-if="isShow" src="https://picsum.photos/510/300?random" width="50%" contain aspect-ratio="1.7" class="ma-4"></v-img>
  </div>
</template>

<script lang="ts" src="./Picture.ts"></script>
