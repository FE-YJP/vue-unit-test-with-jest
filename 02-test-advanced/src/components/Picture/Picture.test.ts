/**
 * 异步测试详解：
 * 1. 定时器：jest.useFakeTimers()->jest.runAllTimers(); 参考 https://jestjs.io/docs/en/jest-object#jestusefaketimers
 * 2. axios请求：参见axios测试详解
 * 3. 其他情况：以上两种情况以外或者以上处理都不能解决的话，建议使用插件flush-promises：https://github.com/kentor/flush-promises
 *
 * 参考：https://vue-test-utils.vuejs.org/zh/guides/#%E6%B5%8B%E8%AF%95%E5%BC%82%E6%AD%A5%E8%A1%8C%E4%B8%BA
 *
 */

import { shallowMount } from '@vue/test-utils'
import Picture from './Picture.vue'

describe('componets.Picture', () => {
  let wrapper: any
  jest.useFakeTimers();

  beforeEach(() => {
    wrapper = shallowMount(Picture)
  })

  describe('snapshot', () => {
    it('图片隐藏时snapshot测试', () => {
      expect(wrapper.vm.$el).toMatchSnapshot();
    })

    it('图片显示时snapshot测试', () => {
      wrapper.setData({ isShow: true })
      expect(wrapper.vm.$el).toMatchSnapshot();
    })
  })

  describe('methods.changeShowLazy', () => {
    it('isShow 值改变', () => {
      wrapper.vm.changeShowLazy()
      jest.runAllTimers();
      expect(wrapper.vm.isShow).toBeTruthy()
    })
  })
})
