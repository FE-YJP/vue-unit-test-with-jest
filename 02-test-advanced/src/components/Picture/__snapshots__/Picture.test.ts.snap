// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`componets.Picture snapshot 图片显示时snapshot测试 1`] = `
<div>
  <v-btn-stub
    activeclass=""
    class="mt-4"
    color="warning"
    depressed="true"
    large="true"
    tag="button"
    type="button"
  >
    
 点击一秒后显示/隐藏图片（定时器测试用例）
 
  </v-btn-stub>
   
  <!---->
</div>
`;

exports[`componets.Picture snapshot 图片隐藏时snapshot测试 1`] = `
<div>
  <v-btn-stub
    activeclass=""
    class="mt-4"
    color="warning"
    depressed="true"
    large="true"
    tag="button"
    type="button"
  >
    
 点击一秒后显示/隐藏图片（定时器测试用例）
 
  </v-btn-stub>
   
  <!---->
</div>
`;
